app/questionnaire/questionnaire_manager.py

```diff
@@ -79,36 +84,17 @@ def process_incoming_answers(self, location, post_data):
 
         return is_valid
 
-    def get_rendering_context(self, location, is_valid=True):
-
-        if is_valid:
-            if location == 'summary':
-                return self.get_summary_rendering_context()
-            else:
-                # apply page answers?
-                self.build_state(location, get_answers(current_user))
-
-        if self.state:
-            self._plumbing_preprocessing()
-            self._conditional_display(self.state)
-
-        # look up the preprocessor and then build the view data
-        return build_questionnaire_model(self._json, self.state)
-
-    def get_summary_rendering_context(self):
-        schema_template_context = get_schema_template_context(self, self._schema)
-        rendered_questionnaire_schema_json = render_template_string(json.dumps(self._json), **schema_template_context)
-
-        # look up the preprocessor and then build the view data
-        return build_summary_model(json.loads(rendered_questionnaire_schema_json))
-
     def build_state(self, item_id, answers):
         # Build the state from the answers
         self.state = None
         if self._schema.item_exists(item_id):
             schema_item = self._schema.get_item_by_id(item_id)
             self.state = schema_item.construct_state()
             self.state.update_state(answers)
+            self._conditional_display(self.state)
+        if self.state:
+            context = build_schema_context(get_metadata(current_user), self._schema.aliases, answers)
+            self.renderer.render_state_item(self.state, context)
```

Comment:
I think we should rename `render_state_item` to `render_state`, we don't refer to state items anywhere else.